File: logica/c/ext.c
#include <stdio.h>

void main(){
    float A, B, C;
    printf("digite o valor do lado A \n");
    scanf("%f", &A);
    printf("digite o valor do lado B \n");
    scanf("%f", &B);
    printf("digite o valor do lado C \n");
    scanf("%f", &C);
    if(A+B > C && A+C>B&& B+C>A){
        if(A==B && B==C){
            printf("O triangulo e equilatero");
        }else{
            if(A != B && A != C && C!=B){
                printf("O triangulo e escaleno");
            }else{
                if(A==B && A!=C || A==C && A!=B || B==C && B != A){
                    printf("O triangulo e isósceles");
                }
            }
        }
    }else{
        printf("triangulo invalido");
    };
  
}

File: logica/python/ext.py
lista = []
lista2 = ["isósceles", "equilatero", "escaleno"]
tipo = 0
for n in range(1, 4):
    lista.append(float(input(f"Digite o valor do lado {n} ")))
if lista[0]+lista[1] > lista[2] and lista[2]+lista[1] > lista[0] and lista[0]+lista[2] > lista[1]:
    for i in range(0, 3):
        for a in range (0,3):
            if lista[i] == lista[a]:
                tipo = tipo + 1/2
    tipo = int(tipo) - 2
    print(f"o triângulo é {lista2[tipo]}")
else: 
  print("triângulo inválido\n")